src/pause_area_map.c

```diff
@@ -119,20 +168,410 @@ static inline void empty_screenbase_23(u32 arg0) {
+                sub_08127304(gUnk_083636D8[r7][0], gUnk_08361220[r5].unk4, gUnk_08361220[r5].unk5,
+                             ARRAY_COUNT(gUnk_083636D8[0][0]));
+                break;
+            }
+    }
+}
+
+static void sub_0812752C(struct AreaMap* areamap) {
+    u8 unk6 = areamap->unk6E0.unk6;
+
+    areamap->unk0.unkA = 0;
+    areamap->unk0.unk18 = 0;
+    areamap->unk0.unk1A = 0;
+    areamap->unk0.unk1C = unk6 + 0xb7;
+    areamap->unk0.unk1E = 0;
+    areamap->unk0.unk20 = 0;
+    areamap->unk0.unk22 = 0;
+    areamap->unk0.unk24 = 0;
+    areamap->unk0.unk26 = 0x20;
+    areamap->unk0.unk28 = 0x16;
+    areamap->unk0.prevScrollX = 0x7fff;
+    areamap->unk0.prevScrollY = 0x7fff;
+    areamap->unk0.paletteOffset = 0;
+    areamap->unk0.unk4 = 0x06004000;  // Charbase 1 -> BG0
+    areamap->unk0.unk2E = 0x18;
+    areamap->unk0.tilemapVram = 0x0600B000;  // Screenbase 22 -> BG0
+    LZ77UnCompVram((u32*)gUnk_08D611C8[unk6], (u32*)0x06004000);
+
+    if (gMainFlags & MAIN_FLAG_BG_PALETTE_TRANSFORMATION_ENABLE)
+        LoadBgPaletteWithTransformation((u16*)(gUnk_08D611F4[unk6] + 0x80), 0x80, 0x80);
+    else {
+        DmaCopy16(3, (u16*)(gUnk_08D611F4[unk6] + 0x80), gBgPalette + 0x80, 0x100);
+        gMainFlags |= MAIN_FLAG_BG_PALETTE_SYNC_ENABLE;
+    }
+
+    sub_08153060(&areamap->unk0);
+}
+
+static void sub_081275F8(struct AreaMap* areamap) {
+    s32 r1;
+
+    switch (areamap->unk48[areamap->unk6E0.unk6]) {
+    case 0:
+    case 1:
+        areamap->unk6E0.unk8 = gUnk_083611F1[areamap->unk6E0.unk6][2] * 4;
+        areamap->unk6E0.unkC = gUnk_083611F1[areamap->unk6E0.unk6][3] * 4;
+        areamap->unk6E0.unk4 = 0x61;
+        break;
+    case 2:
+        r1 = -1;
+        if (areamap->unk120[gUnk_0203AD50].unk28 == areamap->unk6E0.unk6) {
+            r1 = gUnk_0203AD50;
+        }
+        else if (areamap->unk120[gUnk_0203AD3C].unk28 == areamap->unk6E0.unk6) {
+            r1 = gUnk_0203AD3C;
+        }
+
+        if (r1 != -1) {
+            switch (areamap->unk6D0[r1]->unk3) {
+            case 1:
+            case 2:
+                areamap->unk6E0.unk8 = areamap->unk6D0[r1]->unk4 * 8 + 0x10;
+                areamap->unk6E0.unkC = areamap->unk6D0[r1]->unk5 * 8 + 0x10;
+                break;
+            case 0:
+            default:
+                areamap->unk6E0.unk8 = areamap->unk6D0[r1]->unk4 * 8 + 0x8;
+                areamap->unk6E0.unkC = areamap->unk6D0[r1]->unk5 * 8 + 0x8;
+                break;
+            }
+        }
+        else {
+            areamap->unk6E0.unk8 = gUnk_083611F1[areamap->unk6E0.unk6][2] * 4;
+            areamap->unk6E0.unkC = gUnk_083611F1[areamap->unk6E0.unk6][3] * 4;
+        }
+        areamap->unk6E0.unk4 = areamap->unk6E0.unk12;
+        break;
+    }
+}
+
+static void sub_08127760(struct AreaMap* areamap) {
+    if (areamap->unk48[areamap->unk6E0.unk6] == 2) {
+        RLUnCompVram(gUnk_0835EEF8 + gUnk_0835EECC[areamap->unk6E0.unk6], (void*)0x0600c000);
```

Comment:
According to GBATEK `gUnk_0835EEF8` should be 32-bit aligned (and every entry in `gUnk_0835EECC` is a multiple of 4).

Reply:
Yes it works right now but can be easily broken when modding.
I mean you can have an attribute like this https://github.com/jiangzhengwenjz/katam/blob/master/include/palette.h#L8

> Oh, and on that topic, RL decompression into VRAM also wants the destination to be halfword aligned, is it generally better to cast dest to a u16* instead of void*, or does it rather not matter?

It doesn't really matter since the number (address) is explicitly defined by a literal